modules/sdk-coin-stx/src/stx.ts

```diff
@@ -272,4 +302,398 @@ export class Stx extends BaseCoin {
+   * Method to build fungible token transfer transaction
+   * @param builder - fungible token transfer builder
+   * @param contractAddress - token contract address
+   * @param contractName - token contract name
+   * @param accountBalanceData - root address account data
+   * @param pubs - account public keys
+   * @param nonce - account nonce
+   * @param rootAddressDetails - root address details
+   * @param destinationAddressDetails - receive address details
+   * @param stxBalance - native stx balance
+   * @returns {Promise<Transaction>} - built transaction
+   */
+  protected async buildTokenTransferTransaction({
+    builder,
+    contractAddress,
+    contractName,
+    accountBalanceData,
+    pubs,
+    nonce,
+    rootAddressDetails,
+    destinationAddressDetails,
+    stxBalance,
+  }: {
+    builder: FungibleTokenTransferBuilder;
+    contractAddress: string;
+    contractName: string;
+    accountBalanceData: StxAccountBalanceResponse;
+    pubs: string[];
+    nonce: number;
+    rootAddressDetails: AddressDetails;
+    destinationAddressDetails: AddressDetails;
+    stxBalance: string;
+  }): Promise<Transaction> {
+    const txBuilder = builder as FungibleTokenTransferBuilder;
+    const contractTokenName = findContractTokenNameUsingContract(contractAddress, contractName);
+    if (!contractTokenName) {
+      throw new Error('invalid contract address or contract name, not supported');
+    }
+    const assetId = `${contractAddress}.${contractName}::${contractTokenName}`;
+    const tokenBalance = accountBalanceData.fungible_tokens?.[assetId]?.balance;
+    txBuilder.fee({ fee: '200' });
+    txBuilder.numberSignatures(2);
+    txBuilder.fromPubKey(pubs);
+    txBuilder.nonce(nonce);
+    txBuilder.contractAddress(contractAddress);
+    txBuilder.contractName(contractName);
+    if (contractTokenName) {
+      txBuilder.tokenName(contractTokenName);
+    }
+    txBuilder.functionName('transfer');
+    const functionArgs: ClarityValue[] = [
+      uintCV(tokenBalance),
+      standardPrincipalCV(rootAddressDetails.address),
+      standardPrincipalCV(destinationAddressDetails.address),
+    ];
+    if (destinationAddressDetails.memoId) {
+      functionArgs.push(someCV(bufferCVFromString(destinationAddressDetails.memoId)));
+    } else {
+      functionArgs.push(noneCV());
+    }
+    txBuilder.functionArgs(functionArgs);
+    const baseTxn = await txBuilder.build();
+    const txBroadcastFormat = baseTxn.toBroadcastFormat();
+    const txDeserialized = deserializeTransaction(txBroadcastFormat);
+    const serializedHex = serializePayload(txDeserialized.payload).toString('hex');
+    const { recoverableAmount, fee } = await this.getRecoverableAmountAndFee(
+      serializedHex,
+      txBroadcastFormat.length,
+      stxBalance
+    );
+    functionArgs[0] = uintCV(recoverableAmount);
+    txBuilder.functionArgs(functionArgs);
+    txBuilder.fee({ fee: fee });
+    return (await txBuilder.build()) as unknown as Transaction;
```

Comment:
why there was a need to typecast it to unknown first. In the implementation which was done before, this was not needed. Actually why there is a need to typecast it to Transaction